src/modules/event/event.service.ts

```diff
@@ -0,0 +1,93 @@
+import {
+  BadRequestException,
+  Inject,
+  Injectable,
+  NotFoundException,
+} from '@nestjs/common';
+import { EventCreateDto } from './dto/event-create.dto';
+import { EventUpdateDto } from './dto/event-update.dto';
+import { EventCreate } from './domain/event-create';
+import { InjectRepository } from '@nestjs/typeorm';
+import { UserEntity } from '../user/entities/user.entity';
+import { EventEntity } from './entities/event.entity';
+import { Repository } from 'typeorm';
+import { Event } from './domain/event';
+import { EventUpdate } from './domain/event-update';
+
+@Injectable()
+export class EventService {
+  constructor(
+    @InjectRepository(EventEntity)
+    private readonly eventRepository: Repository<EventEntity>,
+  ) {}
+
+  async create(eventCreate: EventCreate): Promise<Event> {
+    this.validateEventTime(eventCreate.timeStart, eventCreate.timeEnd);
+
+    return Event.fromEntity(
+      await this.eventRepository.save(
+        this.eventRepository.create(EventCreate.toEntity(eventCreate)),
+      ),
+    );
+  }
+
+  async findAll(): Promise<Event[]> {
+    return Event.fromEntities(await this.eventRepository.find());
+  }
+
+  async findOne(id: number): Promise<Event> {
+    return Event.fromEntity(await this.findEventOrThrow(id));
+  }
+
+  async update(id: number, updateEvent: EventUpdate): Promise<Event> {
+    this.validateEventTime(updateEvent.timeStart, updateEvent.timeEnd);
+    const eventEntity = await this.findEventOrThrow(id);
```

Comment:
Trước cả 2 cái đều có giá trị thì phải validate trước, nay nó optional thì phải find trước


```suggestion
    const eventEntity = await this.findEventOrThrow(id);
    
    this.validateEventTime(updateEvent.timeStart ?? eventEntity.timeStart, updateEvent.timeEnd ?? eventEntity.timeEnd);
```